cmd/kubeadm/app/cmd/upgrade/upgrade.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package upgrade
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+)
+
+// cmdUpgradeFlags holds the values for the common flags in `kubeadm upgrade`
+type cmdUpgradeFlags struct {
+	kubeConfigPath, cfgPath   string
+	allowExperimentalUpgrades bool
+	allowRCUpgrades           bool
+	printConfig               bool
+}
+
+// NewCmdUpgrade returns the cobra command for `kubeadm upgrade`
+func NewCmdUpgrade(out io.Writer) *cobra.Command {
+	flags := &cmdUpgradeFlags{
+		kubeConfigPath:            "/etc/kubernetes/admin.conf",
+		cfgPath:                   "",
+		allowExperimentalUpgrades: false,
+		allowRCUpgrades:           false,
+		printConfig:               false,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "upgrade",
+		Short: "Upgrade your cluster smoothly to a newer version with this command.",
+		RunE:  subCmdRunE("upgrade"),
+	}
+
+	cmd.PersistentFlags().StringVar(&flags.kubeConfigPath, "kubeconfig", flags.kubeConfigPath, "The KubeConfig file to use for talking to the cluster.")
+	cmd.PersistentFlags().StringVar(&flags.cfgPath, "config", flags.cfgPath, "Path to kubeadm config file (WARNING: Usage of a configuration file is experimental).")
+	cmd.PersistentFlags().BoolVar(&flags.allowExperimentalUpgrades, "allow-experimental-upgrades", flags.allowExperimentalUpgrades, "Show unstable versions of Kubernetes as an upgrade alternative and allow upgrading to an alpha/beta/release candidate versions of Kubernetes.")
```

Comment:
@roberthbailey asked for an explicit flag for this and I kind of agree (we agreed in the meeting as well, I think you were there)